SwanCustomEnvironments/swancustomenvironments/scripts/makenv.sh

```diff
@@ -0,0 +1,292 @@
+    # be called to get it to forget past commands.  Without forgetting
+    # past commands the \$PATH changes we made may not be respected
+    if [ -n \"\${BASH:-}\" -o -n \"\${ZSH_VERSION:-}\" ] ; then
+        hash -r 2> /dev/null
+    fi
+
+    if [ -n \"\${_OLD_VIRTUAL_PS1:-}\" ] ; then
+        PS1=\"\${_OLD_VIRTUAL_PS1:-}\"
+        export PS1
+        unset _OLD_VIRTUAL_PS1
+    fi
+
+    unset VIRTUAL_ENV
+    if [ ! \"\${1:-}\" = \"nondestructive\" ] ; then
+    # Self destruct!
+        unset -f deactivate
+    fi
+}
+
+# unset irrelevant variables
+deactivate nondestructive
+
+VIRTUAL_ENV=\"${ENV_PATH}\"
+export VIRTUAL_ENV
+
+_OLD_VIRTUAL_PATH=\"\$PATH\"
+PATH=\"\$VIRTUAL_ENV/bin:\$PATH\"
+export PATH
+
+# unset PYTHONHOME if set
+# this will fail if PYTHONHOME is set to the empty string (which is bad anyway)
+# could use \`if (set -u; : \$PYTHONHOME) ;\` in bash
+if [ -n \"\${PYTHONHOME:-}\" ] ; then
+    _OLD_VIRTUAL_PYTHONHOME=\"\${PYTHONHOME:-}\"
+    unset PYTHONHOME
+fi
+if [ -n \"\${PYTHONPATH:-}\" ] ; then # added line
+    _OLD_VIRTUAL_PYTHONPATH=\"\${PYTHONPATH:-}\" # added line
+    unset PYTHONPATH # added line
+fi # added line
+
+if [ -z \"\${VIRTUAL_ENV_DISABLE_PROMPT:-}\" ] ; then
+    _OLD_VIRTUAL_PS1=\"\${PS1:-}\"
+    PS1=\"(${ENV_NAME}) \${PS1:-}\"
+    export PS1
+fi
+
+# This should detect bash and zsh, which have a hash command that must
+# be called to get it to forget past commands.  Without forgetting
+# past commands the \$PATH changes we made may not be respected
+if [ -n \"\${BASH:-}\" -o -n \"\${ZSH_VERSION:-}\" ] ; then
+    hash -r 2> /dev/null
+fi
+"
+
+# Create a new bash session to avoid conflicts with the current environment in the background
+env -i bash --noprofile --norc << EOF
+
+export OAUTH2_FILE=${OAUTH2_FILE}
+export OAUTH2_TOKEN=${OAUTH2_TOKEN}
+export KRB5CCNAME=${KRB5CCNAME}
+export KRB5CCNAME_NB_TERM=${KRB5CCNAME_NB_TERM}
+
+if [ -n "$ACCPY_CUSTOM_VERSION" ]; then
+    source ${ACCPY_BASE}/${ACCPY_CUSTOM_VERSION}/setup.sh
+    if [ -d "${ENV_PATH}" ] && [ -n "${CLEAR_ENV}" ]; then
+        rm -rf ${ENV_PATH}
+    fi
+    acc-py venv ${ENV_PATH}
+else
+    if [ -d "${ENV_PATH}" ]; then
+        echo "Recreating (--clear) virtual environment ${ENV_NAME} using Python (${PYTHON_PATH})..."
+    else
+        echo "Creating virtual environment ${ENV_NAME} using Python (${PYTHON_PATH})..."
+    fi
+    ${PYTHON_PATH} -m venv ${ENV_PATH} ${CLEAR_ENV} --copies
+fi
+
+mkdir -p /home/$USER/.local/share/jupyter/kernels
+ln -f -s ${ENV_PATH}/share/jupyter/kernels/${ENV_NAME} /home/$USER/.local/share/jupyter/kernels/${ENV_NAME}
+
+echo "Setting up the virtual environment..."
+source ${ENV_PATH}/bin/activate
+
+# Check if ACCPY_CUSTOM_VERSION is not set and ipykernel is on the requirements file, if not, add the latest version
+if [ -z "$ACCPY_CUSTOM_VERSION" ] && [ -z "$(grep -i 'ipykernel' ${REQ_PATH})" ]; then
+    echo -e "\nipykernel" >> ${REQ_PATH}
+fi
```

Comment:
Before you activate the environment, you can do:
```bash
IPYKERNEL_VERSION=`python -c "import ipykernel; print(ipykernel.__version__)"`
```
so that you get the ipykernel we use in the Python of the image, which is the one that is compatible with our Jupyter server.

Then, here you can do:
```bash
pip install ipykernel==$IPYKERNEL_VERSION
```
to install the right ipykernel.